system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -1145,6 +1152,9 @@ class CLdapClient : public CInterface, implements ILdapClient
 
         createLdapBasedn(NULL, m_ldapconfig->getUserBasedn(), PT_ADMINISTRATORS_ONLY);
         createLdapBasedn(NULL, m_ldapconfig->getGroupBasedn(), PT_ADMINISTRATORS_ONLY);
+                createdOU = true;
```

Comment:
Subsequent code expects the resources to have already been created, so it must not be set until complete